examples/network-firewall/main.tf

```diff
@@ -0,0 +1,125 @@
+provider "aws" {
+  region = local.region
+}
+
+locals {
+  region = "ap-southeast-1"
+}
+
+################################################################################
+# VPC Module
+################################################################################
+
+module "vpc" {
+  source = "../../"
+  cidr   = "10.0.0.0/16"
+  name   = "multiple-nat-firewall"
+  azs    = ["ap-southeast-1a", "ap-southeast-1b", "ap-southeast-1c"]
+
+  enable_nat_gateway     = true
+  one_nat_gateway_per_az = true
+
+  firewall_subnets = [
+    "10.0.3.0/28",
+    "10.0.3.16/28",
+    "10.0.3.32/28",
+  ]
+
+  private_subnets = [
+    "10.0.0.0/24",
+    "10.0.1.0/24",
+    "10.0.2.0/24",
+  ]
+
+  public_subnets = [
+    "10.0.4.0/24",
+    "10.0.5.0/24",
+    "10.0.6.0/24",
+  ]
+
+  # Only uncomment the following line after aws_networkfirewall_firewall has been created on the first run
+  # otherwise it will be a cyclic dependency.
+  # firewall_sync_states = aws_networkfirewall_firewall.example.firewall_status[0].sync_states
+}
+
+################################################################################
+# Network firewall
+################################################################################
+
+resource "aws_networkfirewall_firewall" "example" {
+  name                = "my-network-firewall"
+  firewall_policy_arn = aws_networkfirewall_firewall_policy.example.arn
+  vpc_id              = module.vpc.vpc_id
+
+  dynamic "subnet_mapping" {
+    for_each = module.vpc.firewall_subnets
+    content {
+      subnet_id = subnet_mapping.value
+    }
+  }
+}
```

Comment:
It seems like this two step process exists because the firewall is outside of the module. What if you moved the firewall into the module? You could still keep the policy outside the module.

Full disclosure, I happen to be working on something very similar, so perhaps we should collaborate :)

Reply:
oh hey thanks for the suggestion, i think that's possible, also seeing that the firewall is placed in the `VPC` services on the console, seems fine for it to be created in a `VPC` module as well. 

I didn't think about doing that because i am not sure if this module should be creating a network firewall, but this cyclic dependency does make it seem like it belong together. Should be a small change, i can add the resource soon

Reply:
actually,  i remember why it is not included together now, if i include aws_networkfirewall_firewall, i will also have to include aws_networkfirewall_resource_policy, not sure what is a good base policy that can apply for everyone

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/networkfirewall_firewall#firewall_policy_arn

Reply:
You don't actually need to include the network firewall policy, I pass the arn in as a variable: https://github.com/terraform-aws-modules/terraform-aws-vpc/compare/master...jrstarke:network-firewall?expand=1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R180

I'm going to see if I can make a pull request against your branch